hal/halx86/xbox/reboot.c

```diff
@@ -76,22 +75,31 @@ HalpXboxPowerAction(IN UCHAR Action)
     while (TRUE); /* 'noreturn' function */
 }
 
+DECLSPEC_NORETURN
+VOID
+HalpReboot(VOID)
+{
+    HalpXboxPowerAction(SMC_REG_POWER_RESET);
+}
+
 /* PUBLIC FUNCTIONS **********************************************************/
 
+#ifndef _MINIHAL_
 /*
  * @implemented
  */
 VOID
 NTAPI
-HalReturnToFirmware(IN FIRMWARE_REENTRY Action)
+HalReturnToFirmware(
+    _In_ FIRMWARE_REENTRY Action)
 {
     /* Check what kind of action this is */
     switch (Action)
     {
-        /* All recognized actions */
+        /* All recognized actions: call the internal power function */
+        case HalHaltRoutine:
```

Comment:
I wonder if this addition would make ReactOS turn Xbox off properly - currently it halts on the "it's safe to shutdown" screen, like on a non-ACPI machine.

Reply:
I don't think so, because we don't have any explicit `HalReturnToFirmware(HalHaltRoutine);` call in our code base currently.